cmd/zetaclientd/utils.go

```diff
@@ -97,9 +80,68 @@ func waitForZetacoreToCreateBlocks(ctx context.Context, zc interfaces.ZetacoreCl
 			return fmt.Errorf("zetacore is not ready, timeout %s", time.Since(start).String())
 		}
 
-		logger.Info().Msgf("Failed to get block number, retry : %d/%d", retryCount, attempts)
-		time.Sleep(interval)
+		logger.Info().Msgf("Failed to get block number, retry: %d/%d", retryCount, attempts)
+		return nil
+	}
+
+	return ticker.Run(ctx, interval, task)
+}
+
+// prepareZetacoreClient prepares the zetacore client for use.
+// EXITS THE PROGRAM IF THIS NODE IS NOT AN OBSERVER.
+func prepareZetacoreClient(ctx context.Context, zc *zetacore.Client, cfg *config.Config, logger zerolog.Logger) error {
```

Comment:
consider moving create and prepare functions to a separate file called zetacore_client. 
They don't seem like utility functions